### PR TITLE
Circle ci

### PR DESCRIPTION
**Pull request by cob16**
add circle ci build and workflow

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - setup_remote_docker
+      - run:
+          name: test
+          command: docker-compose run --rm app rspec
+      - run:
+          name: lint
+          command: docker-compose run --rm app rubocop
+  staging_deploy:
+    docker:
+      - image: circleci/buildpack-deps:14.04
```
Comment (cob16) [inline .circleci/config.yml]: these are placeholder steps

File: docker-compose.mount-volume.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+version: '3.4'
+services:
+  app:
+    volumes:
+    - .:/app
```
Comment (cob16) [inline docker-compose.mount-volume.yml]: this makes filesystem mounting optional (as we don't need to use it in the CI)

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: test
+          command: docker-compose run --rm app rspec
+      - run:
+          name: lint
```
Comment (asmega) [inline .circleci/config.yml]: shall we lint before we test?

File: .circleci/config.yml (hunk 1 above)
Comment (asmega) [reply .circleci/config.yml]: 👍

File: .circleci/config.yml (hunk 3 above)
Comment (cob16) [reply .circleci/config.yml]: imo that gets annoying as failing tests are more important than style violations

**Pull request merged**